torch/onnx/_internal/exporter/_capture_strategies.py

```diff
@@ -293,9 +291,17 @@ def forward(self, *_args):
                 self._verbose_print(
                     f"Torch Script model has been saved to '{program_path}'."
                 )
-        return _torchscript_converter.TS2EPConverter(
-            jit_model, flattened_args
-        ).convert()
+        ep = _torchscript_converter.TS2EPConverter(jit_model, flattened_args).convert()
+        if dynamic_shapes is not None:
+            if isinstance(dynamic_shapes, dict):
```

Comment:
Does this fail when the users provide list of dict? If it's flatten inputs issue, we should better try except torch.export.export to capture dynamic shapes error, or simply let it fail. This is and raise only takes care of the partial of the issue.

Reply:
Good point thanks. I will do a try catch instead

Reply:
Turns out this works now

Reply:
removed the try catch and the runtime error